### PR TITLE
Parse union tags in IDL, generate proper openapi

### PR DESCRIPTION
**Pull request by apelisse**
/assign @kwiesmueller @jennybuckley @lavalamp 
/cc @sttts @jpbetz

### CODE REVIEW

File: pkg/generators/openapi.go [hunk 1]
```diff
@@ -416,13 +423,13 @@ func (g openAPITypeWriter) generateMemberExtensions(m *types.Member, parent *typ
 			klog.V(2).Infof("%s %s\n", errorPrefix, e)
 		}
 	}
-	g.emitExtensions(extensions)
+	g.emitExtensions(extensions, nil)
 	return nil
 }
 
-func (g openAPITypeWriter) emitExtensions(extensions []extension) {
+func (g openAPITypeWriter) emitExtensions(extensions []extension, unions []union) {
```
Comment (jennybuckley) [inline pkg/generators/openapi.go]: Is there a reason why unions isn't just a part of extensions?

Comment (apelisse) [reply pkg/generators/openapi.go]: Yeah, I would have to completely change the way extensions are built (they're meant for list of scalars) and I didn't want to make the type worse by adding a bunch of unrelated things.

Comment (jennybuckley) [reply pkg/generators/openapi.go]: sounds good

**Pull request merged**